### ISSUE TITLE
Notion import error "Folder already exists"

### ISSUE THREAD

**Issue by Sinled** [obsidianmd/obsidian-importer#83]
Hi, there is an error with import plugin trying create folder that already exist, without any meaningful stack trace.

```
app.js:1 Uncaught (in promise) Error: Folder already exists.
    at t.<anonymous> (app.js:1:1024123)
    at app.js:1:236930
    at Object.next (app.js:1:237035)
    at a (app.js:1:235753)
```

I've downloaded and built plugin from code, and it seems problem is here in `format-importer.ts`
```ts
async createFolders(path: string): Promise<TFolder> {
    // ...
    let folder = this.vault.getAbstractFileByPath(normalizedPath);
    // ...
    await this.vault.createFolder(normalizedPath);
```

`getAbstractFileByPath` returns `null` if folder exist but have different case in name.
and when `createFolder` is trying to create folder but with different case - obsidian throws error.
I've replaced locally `getAbstractFileByPath` with `getAbstractFileByPathInsensitive` and after that it seems import works fine, but i'm not sure if this might break in some other place.

Also there is another bug - import won't do anything if notion export archive has nested archive with `*-Part-1.zip` in it.
In such case only one attachement folder with "part" archive in it created

**Comment by lishid** [comment]
> Also there is another bug - import won't do anything if notion export archive has nested archive with *-Part-1.zip in it.
In such case only one attachement folder with "part" archive in it created

I've removed the requirement for detecting the notion ID from the zip file - that should fix this part.

**Comment by joshuatazrein** [comment]
Would it be possible to message me a `.zip` of your data so I can work on debugging? You can share via Discord @joshua.palindrome or share with `joshuatreinier@gmail.com`

**Comment by Sinled** [comment]
@joshuatazrein There are a lot of personal data, so i can't send it. I've tried to make smaller reproduction but for smaller export notion won't make nested archive. After that i've tried to make smaller archive with test data, but i cant find way to make archive nested directly in archive - mac os don't allow it, running `zip` from terminal on zip file also gives not result.

**Comment by joshuatazrein** [comment]
Understandable! Yeah I'm having trouble finding nested zips because I'm on a Mac.

**Comment by joshuatazrein** [comment]
I think the `createFolders` bug should be fixed in `main`, since that would potentially affect other plugins.

**Comment by joshuatazrein** [comment]
I believe this has been fixed by the change in `[createFolders](https://github.com/obsidianmd/obsidian-importer/blob/1dfe37d8635d65a21902006b443a3637673c0bc0/src/format-importer.ts#L151C5-L151C5)`.

**Comment by kepano** [comment]
Closed via [1.5.5](https://github.com/obsidianmd/obsidian-importer/releases/tag/1.5.5)
